diff -rupN vtk-5.6.1_allPatchesButAero/GUISupport/Qt/QVTKWidget.cxx vtk-5.6.1/GUISupport/Qt/QVTKWidget.cxx
--- vtk-5.6.1_allPatchesButAero/GUISupport/Qt/QVTKWidget.cxx	Mon Sep 27 14:33:34 2010
+++ vtk-5.6.1/GUISupport/Qt/QVTKWidget.cxx	Fri Feb  8 08:44:38 2013
@@ -53,6 +53,11 @@
 #include "qx11info_x11.h"
 #endif
 
+#if defined(Q_WS_WIN)
+#include <windows.h>
+#include <QSysInfo>
+#endif
+
 #include "vtkstd/map"
 #include "vtkInteractorStyleTrackballCamera.h"
 #include "vtkRenderWindow.h"
@@ -1367,6 +1372,22 @@ void QVTKWidget::x11_setup_window()
 
 #endif
 }
+
+#if defined(Q_WS_WIN)
+bool QVTKWidget::winEvent(MSG* msg, long*)
+{
+  // Starting with Windows Vista, Microsoft introduced WDDM.
+  // We need to call InvalidateRect() to work with WDDM correctly,
+  // especially when AERO is off.
+
+  if(msg->message == WM_PAINT &&
+     QSysInfo::windowsVersion() >= QSysInfo::WV_VISTA)
+  {
+    InvalidateRect(this->winId(), NULL, FALSE);
+   }
+   return false;
+}
+#endif
 
 #if defined (QVTK_USE_CARBON)
 OSStatus QVTKWidget::DirtyRegionProcessor(EventHandlerCallRef, EventRef event, void* wid)
diff -rupN vtk-5.6.1_allPatchesButAero/GUISupport/Qt/QVTKWidget.h vtk-5.6.1/GUISupport/Qt/QVTKWidget.h
--- vtk-5.6.1_allPatchesButAero/GUISupport/Qt/QVTKWidget.h	Mon Sep 27 14:33:34 2010
+++ vtk-5.6.1/GUISupport/Qt/QVTKWidget.h	Mon Oct  1 10:55:55 2012
@@ -240,6 +240,10 @@ protected:
   // that VTK chooses
   void x11_setup_window();
 
+#if defined(Q_WS_WIN)
+  bool winEvent(MSG* msg, long* result);
+#endif
+
 #if defined(QVTK_USE_CARBON)
   EventHandlerUPP DirtyRegionHandlerUPP;
   EventHandlerRef DirtyRegionHandler;
