diff --git a/VTK-6.1.0-base/GUISupport/Qt/QVTKWidget.cxx b/VTK-6.1.0/GUISupport/Qt/QVTKWidget.cxx
index abd24e7..07369eb 100644
--- a/VTK-6.1.0-base/GUISupport/Qt/QVTKWidget.cxx
+++ b/VTK-6.1.0/GUISupport/Qt/QVTKWidget.cxx
@@ -48,7 +48,7 @@
 #include "qx11info_x11.h"
 #endif
 
-#if defined(Q_WS_WIN)
+#if defined(Q_OS_WIN)
 # include <windows.h>
 # include <QSysInfo>
 #endif
@@ -812,16 +812,20 @@ void QVTKWidget::x11_setup_window()
 #endif
 }
 
-#if defined(Q_WS_WIN)
-bool QVTKWidget::winEvent(MSG* msg, long*)
+#if defined(Q_OS_WIN)
+bool QVTKWidget::nativeEvent(const QByteArray& eventType, void* message, long*)
 {
   // Starting with Windows Vista, Microsoft introduced WDDM.
   // We need to call InvalidateRect() to work with WDDM correctly,
   // especially when AERO is off.
-  if(msg->message == WM_PAINT &&
-    QSysInfo::windowsVersion() >= QSysInfo::WV_VISTA)
+  if (eventType == "windows_generic_MSG")
     {
-    InvalidateRect(this->winId(), NULL, FALSE);
+    MSG* msg = (MSG*)message;
+    if(msg->message == WM_PAINT &&
+      QSysInfo::windowsVersion() >= QSysInfo::WV_VISTA)
+      {
+      InvalidateRect((HWND)this->winId(), NULL, FALSE);
+      }
     }
   return false;
 }
diff --git a/VTK-6.1.0-base/GUISupport/Qt/QVTKWidget.h b/VTK-6.1.0/GUISupport/Qt/QVTKWidget.h
index 631d300..f396e73 100644
--- a/VTK-6.1.0-base/GUISupport/Qt/QVTKWidget.h
+++ b/VTK-6.1.0/GUISupport/Qt/QVTKWidget.h
@@ -248,8 +248,8 @@ protected:
   // that VTK chooses
   void x11_setup_window();
 
-#if defined(Q_WS_WIN)
-  bool winEvent(MSG* msg, long* result);
+#if defined(Q_OS_WIN)
+  virtual bool nativeEvent(const QByteArray& eventType, void* message, long* result);
 #endif
 
 #if defined(QVTK_USE_CARBON)
