--- a/Source/DataStructureAndEncodingDefinition/gdcmReader.cxx
+++ b/Source/DataStructureAndEncodingDefinition/gdcmReader.cxx
@@ -671,11 +671,13 @@ bool Reader::InternalReadCommon(const T_Caller &caller)
         {
         gdcmDebugMacro( "No way this is DICOM" );
         success = false;
+        throw ex;
         }
       }
 #else
     gdcmDebugMacro( ex.what() ); (void)ex;
     success = false;
+    throw ex;
 #endif /* GDCM_SUPPORT_BROKEN_IMPLEMENTATION */
     }
   catch( Exception &ex )
@@ -683,11 +685,13 @@ bool Reader::InternalReadCommon(const T_Caller &caller)
     (void)ex;  //to avoid unreferenced variable warning on release
     gdcmDebugMacro( ex.what() );
     success = false;
+    throw ex;
     }
   catch( ... )
     {
     gdcmWarningMacro( "Unknown exception" );
     success = false;
+    throw;
     }
 
   //if( success ) assert( Stream->eof() );
@@ -698,11 +702,13 @@ bool Reader::InternalReadCommon(const T_Caller &caller)
     (void)ex;  //to avoid unreferenced variable warning on release
     gdcmDebugMacro( ex.what() );
     success = false;
+    throw ex;
     }
   catch( ... )
     {
     gdcmWarningMacro( "Unknown exception" );
     success = false;
+    throw;
     }
 
   // FIXME : call this function twice...
