--- gdcm-2.0.19/Source/DataStructureAndEncodingDefinition/gdcmReader.cxx	Tue Jun 12 12:16:11 2012
+++ gdcm-2.0.19_with_modifications/Source/DataStructureAndEncodingDefinition/gdcmReader.cxx	Tue Jun 12 11:39:31 2012
@@ -657,6 +657,7 @@
 #else
     gdcmDebugMacro( ex.what() );
     success = false;
+    throw ex;
 #endif /* GDCM_SUPPORT_BROKEN_IMPLEMENTATION */
     }
   catch( Exception &ex )
@@ -664,11 +665,13 @@
 	(void)ex;  //to avoid unreferenced variable warning on release
     gdcmDebugMacro( ex.what() );
     success = false;
+    throw ex;
     }
   catch( ... )
     {
     gdcmWarningMacro( "Unknown exception" );
     success = false;
+    throw;
     }
 
   //if( success ) assert( Stream->eof() );
@@ -679,11 +682,13 @@
 	(void)ex;  //to avoid unreferenced variable warning on release
     gdcmDebugMacro( ex.what() );
     success = false;
+    throw ex;
     }
   catch( ... )
     {
     gdcmWarningMacro( "Unknown exception" );
     success = false;
+    throw;
     }
   //  if( !success )
   //    {
