diff -uprN gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx
--- gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx	2011-01-25 16:48:14.000000000 +0100
+++ gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmJPEG2000Codec.cxx	2011-06-17 14:22:34.000000000 +0200
@@ -310,6 +310,7 @@ bool JPEG2000Codec::Decode(DataElement c
     is.write(buffer, totalLen);
     delete[] buffer;
     std::stringstream os;
+    os.exceptions( std::ostream::failbit | std::ostream::badbit );
     bool r = Decode(is, os);
     if(!r) return false;
     out = in;
diff -uprN gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmJPEGCodec.cxx gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmJPEGCodec.cxx
--- gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmJPEGCodec.cxx	2011-01-25 16:48:14.000000000 +0100
+++ gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmJPEGCodec.cxx	2011-06-17 14:21:38.000000000 +0200
@@ -154,6 +154,7 @@ bool JPEGCodec::Decode(DataElement const
   const ByteValue *jpegbv = in.GetByteValue();
   if( !sf && !jpegbv ) return false;
   std::stringstream os;
+  os.exceptions( std::ostream::failbit | std::ostream::badbit );
   if( sf )
     {
     //unsigned long pos = 0;
diff -uprN gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmRAWCodec.cxx gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmRAWCodec.cxx
--- gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmRAWCodec.cxx	2011-01-25 16:48:14.000000000 +0100
+++ gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmRAWCodec.cxx	2011-06-17 14:21:59.000000000 +0200
@@ -98,6 +98,7 @@ bool RAWCodec::DecodeBytes(const char* i
   std::stringstream is;
   is.write(inBytes, inBufferLength);
   std::stringstream os;
+  os.exceptions( std::ostream::failbit | std::ostream::badbit );
   bool r = Decode(is, os);
   assert( r );
   if(!r) return false;
diff -uprN gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmRLECodec.cxx gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmRLECodec.cxx
--- gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmRLECodec.cxx	2011-01-25 16:48:14.000000000 +0100
+++ gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmRLECodec.cxx	2011-06-17 14:23:15.000000000 +0200
@@ -568,6 +568,7 @@ bool RLECodec::Decode(DataElement const 
     sf->WriteBuffer( is );
     SetLength( len );
       std::stringstream os;
+      os.exceptions( std::ostream::failbit | std::ostream::badbit );
       bool r = Decode(is, os);
       assert( r == true );
       (void)r; //warning removal
--- gdcm-2.0.17_original/Source/MediaStorageAndFileFormat/gdcmJPEGCodec.cxx	2011-01-25 16:48:14.000000000 +0100
+++ gdcm-2.0.17/Source/MediaStorageAndFileFormat/gdcmJPEGCodec.cxx	2011-06-22 16:58:41.000000000 +0200
@@ -339,6 +339,7 @@ bool JPEGCodec::Code(DataElement const &
 bool JPEGCodec::Decode(std::istream &is, std::ostream &os)
 {
   std::stringstream tmpos;
+  tmpos.exceptions( std::ostream::failbit | std::ostream::badbit );
   if ( !Internal->Decode(is,tmpos) )
     {
 #ifdef GDCM_SUPPORT_BROKEN_IMPLEMENTATION
 