diff -upr gdcm-2.4.3/Source/MediaStorageAndFileFormat/gdcmImageCodec.cxx gdcm-2.4.3-patched/Source/MediaStorageAndFileFormat/gdcmImageCodec.cxx
--- gdcm-2.4.3/Source/MediaStorageAndFileFormat/gdcmImageCodec.cxx	2014-07-26 17:19:49.000000000 +0200
+++ gdcm-2.4.3-patched/Source/MediaStorageAndFileFormat/gdcmImageCodec.cxx	2014-08-06 12:22:31.626422000 +0200
@@ -406,30 +406,50 @@ bool ImageCodec::DoOverlayCleanup(std::i
       nmask = (int16_t)(nmask >> ( PF.GetBitsAllocated() - PF.GetBitsStored() - 1 ));
 
       uint16_t c;
-      while( is.read((char*)&c,2) )
-        {
-        c = (uint16_t)(c >> (PF.GetBitsStored() - PF.GetHighBit() - 1));
-        if ( c & smask )
-          {
-          c = (uint16_t)(c | nmask);
-          }
-        else
+      try
+      {
+        while( is.read((char*)&c,2) )
           {
-          c = c & pmask;
+          c = (uint16_t)(c >> (PF.GetBitsStored() - PF.GetHighBit() - 1));
+          if ( c & smask )
+            {
+            c = (uint16_t)(c | nmask);
+            }
+          else
+            {
+            c = c & pmask;
+            }
+          os.write((char*)&c, 2 );
           }
-        os.write((char*)&c, 2 );
+      }
+      catch (std::ios_base::failure &f)
+      {
+        if (!is.eof())
+        {
+          throw f;
         }
       }
+      }
     else // Pixel are unsigned
       {
 #if 1
       uint16_t c;
-      while( is.read((char*)&c,2) )
+      try
+      {
+        while( is.read((char*)&c,2) )
+          {
+          c = (uint16_t)(
+            (c >> (PF.GetBitsStored() - PF.GetHighBit() - 1)) & pmask);
+          os.write((char*)&c, 2 );
+          }
+      }
+      catch (std::ios_base::failure &f)
+      {
+        if (!is.eof())
         {
-        c = (uint16_t)(
-          (c >> (PF.GetBitsStored() - PF.GetHighBit() - 1)) & pmask);
-        os.write((char*)&c, 2 );
+          throw f;
         }
+      }
       //os.rdbuf( is.rdbuf() );
 #else
       //std::ostreambuf_iterator<char> end_of_stream_iterator;
