--- a/dcmimgle/libsrc/dimomod.cc
+++ b/dcmimgle/libsrc/dimomod.cc
@@ -279,6 +279,8 @@ void DiMonoModality::determineRepresenta
     UsedBits = DicomImageClass::rangeToBits(MinValue, MaxValue);
     if ((docu != NULL) && (docu->getFlags() & CIF_UseFloatingInternalRepresentation))
         Representation = EPR_float32;
+    else if ((docu != NULL) && (Rescaling && RescaleSlope > -1.0 && RescaleSlope < 1.0))
+        Representation = EPR_float32;
     else if ((docu != NULL) && (docu->getFlags() & CIF_UseAbsolutePixelRange))
         Representation = DicomImageClass::determineRepresentation(AbsMinimum, AbsMaximum);
     else
